src/index.ts

```diff
@@ -231,6 +233,70 @@ function launchMetals(
       );
     }
 
+    registerCommand("metals.new-scala-file", async () => {
+      const currentDoc = await workspace.document;
+      const currentPath = currentDoc.uri;
+      const parentDir = path.dirname(currentPath);
+
+      const fileOptions = [
+        {
+          kind: "class",
+          value: 0,
+          label: "Class"
+        },
+        {
+          kind: "object",
+          value: 1,
+          label: "Object"
+        },
+        {
+          kind: "trait",
+          value: 2,
+          label: "Trait"
+        },
+        {
+          kind: "package-object",
+          value: 3,
+          label: "Package Object"
+        },
+        {
+          kind: "worksheet",
+          value: 4,
+          label: "Worksheet"
+        }
+      ];
+
+      const fileSelection = await workspace.showQuickpick(
+        fileOptions.map(option => option.label),
+        "Select the kind of file to create"
+      );
+
+      const desiredFileType =
+        fileOptions.find(option => fileSelection === option.value)?.kind || "";
```

Comment:
Doing this and then checking for an empty string below isn't probably too spectacular. I wanted a `Fold` after this `find`, but yea... I'm not sure on the best way to either get this value and do something with it or just return, so instead I just forced it into a string, and then checked it below. Advice on how to clean this up?

Reply:
use `Option` from `fp-ts` and related methods :P

More concretely, why not leave it as `string | undefined` and check for undefined in the if?